src/components/editor/QEditor.vue

```diff
@@ -0,0 +1,521 @@
+			v-html="sanitize(initialContent)"
+		></div>
+		<EditorActions
+			v-show="toggleAddContent"
+			class="absolute modal-animation"
+			:style="`top:` + this.addContentPosTop + `px;` + `left:` + this.addContentPosLeft + `px`"
+			@image="actionsUpload"
+		/>
+		<div
+			v-if="waitingImage"
+			class="absolute w-11/12 h-44 bg-lightInput dark:bg-gray7 rounded-lg animate-pulse flex justify-center items-center"
+			:style="`top:` + this.addContentPosTop + `px`"
+		>
+			<p class="text-sm text-gray5 dark:text-gray3">Uploading image...</p>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import DOMPurify from 'dompurify'
+import Vue from 'vue'
+import type { RangeStatic, Quill } from 'quill'
+import type { PropType } from 'vue'
+import QuillMarkdown from 'quilljs-markdown'
+import hljs from 'highlight.js'
+import turndownService from './TurndownService'
+import EditorActions from './EditorActions.vue'
+import {
+	createEditorImageSet,
+	counterModuleFactory,
+	ImageBlotFactory,
+	EditorImages,
+} from '@/pages/post/quillExtensions'
+
+interface IData {
+	toggleAddContent: boolean
+	addContentPosTop: number
+	waitingImage: boolean
+	qeditor: Quill | null
+	editor: Quill | null
+	addContentPosLeft: number
+	editorImages: EditorImages | null
+}
+
+const toolbarOptions = [
+	[`bold`, `italic`, `underline`, `strike`],
+	[`blockquote`, `code-block`, `link`],
+	[{ header: 2 }],
+	[{ list: `ordered` }, { list: `bullet` }],
+]
+
+const options = {
+	placeholder: `Start typing here...`,
+	readOnly: false,
+	theme: `bubble`,
+	bounds: `#editor`,
+	scrollingContainer: `#editor`,
+	modules: {
+		syntax: {
+			highlight: (code: string) => hljs.highlightAuto(code).value,
+		},
+		counter: true,
+		toolbar: {
+			container: toolbarOptions,
+		},
+	},
+}
+
+export default Vue.extend({
+	components: { EditorActions },
+	props: {
+		initialContent: {
+			type: String,
+			required: true,
+		},
+		initialEditorImages: {
+			type: Map as PropType<Map<string, { key: string; counter: string } | {}>>,
+			default: {},
+		},
+		validImageTypes: {
+			type: Array as PropType<string[] | undefined>,
+			required: true,
+		},
+		imageUploader: {
+			type: Function as PropType<(file: File, encrypt?: boolean) => Promise<any>>,
+			required: true,
+		},
+		isPrimaryWidget: {
+			type: Boolean,
+			default: false,
+		},
+		allowedTags: {
+			type: Array as PropType<string[] | undefined>,
+			default: () => [],
+		},
+		maxPostImages: {
+			type: Number,
+			default: 10,
+		},
+		encryptedContent: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	data(): IData {
+		return {
+			toggleAddContent: false,
+			addContentPosTop: 0,
+			waitingImage: false,
+			qeditor: null,
+			editor: null,
+			addContentPosLeft: 0,
+			editorImages: new Map(),
+		}
+	},
+	mounted() {
+		if (this.initialEditorImages) {
+			this.editorImages = this.initialEditorImages as any
+		}
+		this.setupEditor()
+	},
+	methods: {
+		getContentImages(content: string) {
```

Comment:
Functions that don't use any reference to "this" could be separated out to a helpers.ts file for the editor.

Reply:
(there are lots of these examples)